R/utilities-mpi-error-handling.R

```diff
@@ -0,0 +1,153 @@
+loadSimulationOnCores <- function(structureSet, logFolder = getwd()) {
+  success <- FALSE
+  Rmpi::mpi.bcast.Robj2slave(obj = structureSet)
+  Rmpi::mpi.remote.exec(sim <- loadSimulationWithUpdatedPaths(structureSet$simulationSet))
+  simulationLoaded <- Rmpi::mpi.remote.exec(checkSimulationLoaded(simulation = sim, simulationFilePath = structureSet$simulationSet$simulationFile))
+  success <- checkAllCoresSuccessful(simulationLoaded)
+  validateIsLogical(success)
+  if (success) {
+    logWorkflow(message = paste("Simulation file", structureSet$simulationSet$simulationFile, "loaded successfully on all cores"), pathFolder = logFolder)
+  } else {
+    logErrorThenStop(message = paste("Simulation file", structureSet$simulationSet$simulationFile, "not loaded successfully on all cores"), logFolderPath = logFolder)
+  }
+}
+
+#' @title loadSimulationOnCores
+#' @description Send libraryName to core, load the library and check that it has been loaded successfully
+#' @param libraryName string containing name of library to be loaded
+loadLibraryOnCores <- function(libraryName, logFolder = getwd()) {
+  success <- FALSE
+  Rmpi::mpi.bcast.Robj2slave(obj = libraryName)
+  Rmpi::mpi.remote.exec(library(libraryName, character.only = TRUE))
+  libraryLoaded <- Rmpi::mpi.remote.exec(checkLibraryLoaded(libraryName))
+  success <- checkAllCoresSuccessful(libraryLoaded)
+  validateIsLogical(success)
+  if (success) {
+    logWorkflow(message = paste(libraryName, "loaded successfully on all cores"), pathFolder = logFolder)
+  } else {
+    logErrorThenStop(message = paste(libraryName, "not loaded successfully on all cores"), logFolderPath = logFolder)
+  }
+}
+
+#' @title loadSimulationOnCores
+#' @description Send libraryName to core, load the library and check that it has been loaded successfully
+#' @param libraryName string containing name of library to be loaded
+updateIndividualParametersOnCores <- function(individualParameters, logFolder = getwd()) {
+  success <- FALSE
+  Rmpi::mpi.bcast.Robj2slave(obj = individualParameters)
+  individualParametersUpdated <- Rmpi::mpi.remote.exec(updateSimulationIndividualParameters(simulation = sim, individualParameters))
+  success <- checkAllCoresSuccessful(individualParametersUpdated)
+  validateIsLogical(success)
+  if (success) {
+    logWorkflow(message = "Individual parameters updated successfully on all cores.", pathFolder = logFolder)
+  } else {
+    logErrorThenStop(message = "Individual parameters updated successfully on all cores.", logFolderPath = logFolder)
+  }
+}
+
+
+#' @title verifySimulationRunSuccessful
+#' @description Check that all cores ran simulation successfully
+#' @param simulationRunSuccess logical vector indicating success of simulation run on all cores
+verifySimulationRunSuccessful <- function(simulationRunSuccess, logFolder = getwd()) {
+  success <- FALSE
```

Comment:
No need to initialize to FALSE since it's reset right after

Reply:
Removed